tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -275,6 +275,20 @@ func (a *AwsProvider) getDecodedPassword(instanceID string) ([]byte, error) {
 
 }
 
+// DestroyWindowsVM destroys the given Windows VM
+func (a *AwsProvider) DestroyWindowsVM(instanceID string) error {
+	var err error
+	if err = a.TerminateInstance(instanceID); err != nil {
+		return fmt.Errorf("failed deleting the instance %s: %v", instanceID, err)
+	}
+	if err := a.waitUntilInstanceTerminated(instanceID); err != nil {
+		// As of now, I want the error thrown to be consistent across all cloud providers, so I am not specializing
+		// the information in the error message string but the err will have more context
+		return fmt.Errorf("failed deleting the instance %s: %v", instanceID, err)
+	}
```

Comment:
This can be made into a helper that is also called from `DestroyWindowsVMs()`

Reply:
We've a fire & forget operation within DestrotWindowsVMs method, so we cannot share the code.